modules/exploits/linux/http/eyesofnetwork_autodiscovery_rce.rb

```diff
@@ -71,29 +100,149 @@ def server_addr
 
   def check
     vprint_status("Running check")
-    res = send_request_cgi 'uri' => normalize_uri(target_uri.path, '/eonapi/getApiKey')
+    res = send_request_cgi 'uri' => normalize_uri(target_uri.path,'/login.php')
 
     unless res
       return CheckCode::Unknown('Connection failed')
     end
 
-    unless res.code == 401 && res.body.include?('api_version')
+    unless res.code == 200 && res.body.include?('EyesOfNetwork')
       return CheckCode::Safe('Target is not an EyesOfNetwork application.')
     end
 
-    version = res.get_json_document()['api_version'] rescue ''
+    res_api = send_request_cgi 'uri' => normalize_uri(target_uri.path, '/eonapi/getApiKey')
+
+    unless res_api
+      return CheckCode::Unknown('Connection failed')
+    end
+
+    if res_api.code == 404 #The EON api was only introduced in EON 5.2
+      @version = '5.1'
+      return CheckCode::Appears("Target is EyesOfNetwork version 5.1 or older.")
```

Comment:
This condition is a little too loose. So far all you've done is found `'EyesOfNetwork'` in the body of the `login.php` resource and determined that `/eonapi/getApiKey` does not exist. Is there any way you can determine with a bit more certainty that the application is in fact installed on the target? Maybe a more thorough check of 'login.php'? Another resource you could request?